### PR TITLE
Feature/User CRUD and session management

### PR DESCRIPTION
**Pull request by AlexPop28**
Closes https://trello.com/c/M3NW9FH5
Closes https://trello.com/c/1V97RvWu

### CODE REVIEW

**Review by Laurcons — APPROVED** [review]
Clean code, componentized.

File: src/App.tsx [hunk 1]
```diff
@@ -14,7 +19,7 @@ const router = createBrowserRouter([
   },
   {
     path: "/login",
-    element: <LoginPage />,
+    element: <Login />,
```
Comment (Laurcons) [inline src/App.tsx]: Please append Page to components that act as pages

File: src/components/UserForm.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  role?: string;
+  setRole?: Dispatch<SetStateAction<string>>;
+  status?: string;
+  setStatus?: Dispatch<SetStateAction<string>>;
+  onClick: () => Promise<void>;
+  buttonText: string;
+  children?: React.ReactNode;
+}
+
+const UserForm: FC<UserFormProps> = ({
```
Comment (Laurcons) [inline src/components/UserForm.tsx:22]: Have you considered using Formik? https://formik.org/docs/overview See examples in my MPP project

**Review by Laurcons — CHANGES_REQUESTED** [review]

File: src/components/UserForm.tsx (hunk 2 above)
Comment (AlexPop28) [reply src/components/UserForm.tsx:22]: I use it for destination forms in the other branch. I will refactor this one as well to use it.

File: src/lib/hooks/useUser.tsx [hunk 3]
```diff
@@ -1,6 +1,56 @@
...
+        .catch(err => {
+          if (err instanceof AxiosError) {
+            if (err.response?.status === 403) return;
+          }
+          console.log(err);
+        })
+  );
+
+  const invalidate = () => {
+    queryClient.invalidateQueries([role, "users", "me"]);
```
Comment (Laurcons) [inline src/lib/hooks/useUser.tsx]: Do not add the role as a key, the user cannot be logged into two roles at the same time so there needs to be a single key referencing a "/users/me" endpoint. You invalidate the entire ['users', 'me] key, regardless of the role. I don't think this affects behavior too much, though.

**Review by Laurcons — APPROVED** [review]

File: src/lib/hooks/useUser.tsx (hunk 3 above)
Comment (AlexPop28) [reply src/lib/hooks/useUser.tsx]: There is something weird happening for logout. No matter what I tried it seems that after the query is invalidated the re-fetching is done with the token set anyways (even though it is explicitly removed in the `logout` function). Do you have any ideas for a fix? I tried creating the `axios` object in the `queryFn` but I think that the whole function might be cached the way it is at the time of creation and until a rerender of `UserProvider` happens (no idea when that is but it takes a few seconds) it doesn't perform the intended query.

**Pull request merged**